### DISCUSSION TITLE
What is the best way to automate commits from your computer?

### DISCUSSION THREAD

**Discussion by LEDparty** [community/community#129045] [Repositories]
I'm wondering because I tend to find the process of updating a repository kind of annoying and frustrating. I made this script, but some folks on reddit strongly disapproved of it for security reasons:

https://github.com/LEDparty/lazy-commit

I'll admit: I didn't entirely understand why they were downvoting and complaining. One of the reasons was that with "git add .", you might add files you didn't intend to. I think that criticism is kinda strange because I've always felt it's best to have a totally separate folder for the respo files. I did include safegaurds against those types of errors. 

A professional coder friend of mine showed me how in visual basic there is way to make changes to your github repository through VSCode, so I thought that maybe there is something inside of gitkraken that could make commits even more effortless than my script. After all, each little change still requires me to:

-specify whether or not it's an initialization

-if yes, then copy the repo URL

-enter my username at the terminal

-copy and paste the SSH key from the github website

I'm mostly just wondering what my options are. I tend to assume that if something I do gets heavy criticism, then there's at least some reasoning behind it, yet I didn't really understand. I would think if no one else is participating in my repositories, then there can't be much of a problem just forcing my changes every time.

**Comment by FreddyMSchubert** [comment]
I usually just do this:
`git config --global alias.imlazy '!git add -A && git commit -m "${1-Commit all changes}" && git push'`
Then you can easily make & push your commits like this:
`git imlazy "Your specific commit message"`
I don't think much automization beyond that point is overly useful (like, how many repos are you making? why do you need to get the ssh key every time?)
hope that helps!

**Reply by LEDparty** [reply]
So is "alias.imlazy" a name you are creating for a git command alias?

**Reply by FreddyMSchubert** [reply]
> So is "alias.imlazy" a name you are creating for a git command alias? 

Yes, exactly! 
You could probably do the same thing with .zshrc if you're more comfortable with that. Just adds a custom command, basically.

**Comment by ebndev** [comment]
Hi @LEDparty, thanks for posting in GitHub Discussions! 

I've gone ahead and moved this post to our Repositories category, as it may be more relevant there 😄
